index.js

```diff
@@ -1,300 +1,147 @@
-    const d = buf.slice(2)
-    const b64part = d.toString('base64')
-    return b64part + '.sig.ed25519'
-  },
+  if (Buffer.isBuffer(input)) {
+    if (input.length < 2)
+      throw new Error('Buffer is missing first two type & format bytes')
 
-  box(buf) {
-    const tf = buf.slice(0, 2)
-    const d = buf.slice(2)
-    if (tf.equals(BOX1_TF)) return d.toString('base64') + '.box'
-    else if (tf.equals(BOX2_TF)) return d.toString('base64') + '.box2'
-    else throw new Error('Unknown box: ' + buf)
-  },
+    if (input.equals(NIL.TFD)) return null
+    if (input.slice(0, 2).equals(BOOL.TF)) return decoder.bool(input)
+    if (input.slice(0, 2).equals(STRING.TF))
+      return input.slice(2).toString('utf8')
 
-  string(buf) {
-    const d = buf.slice(2)
-    return d.toString('utf8')
-  },
+    const type = TYPES.find((type) => type.code.equals(input.slice(0, 1)))
+    if (type) {
+      const format = type.formats.find((format) =>
+        format.code.equals(input.slice(1, 2))
+      )
+      if (format) {
+        if (type.sigil || format.suffix)
+          return decoder.classic(input, type, format)
+      }
+    }
 
-  boolean(buf) {
-    const d = buf.slice(2)
-    return d.equals(BOOL_TRUE)
-  },
-}
+    throw new Error('unknown type/ format')
+  }
 
-function decode(input) {
-  if (Array.isArray(input)) {
-    return input.map(decode)
-  } else if (Buffer.isBuffer(input)) {
-    if (input.length < 2)
-      throw new Error(
-        'Buffer is missing first two type&format fields: ' + input
-      )
-    const t = input.slice(0, 1)
-    const tf = input.slice(0, 2)
-    if (tf.equals(STRING_TF)) return decoder.string(input)
-    else if (tf.equals(BOOL_TF)) return decoder.boolean(input)
-    else if (tf.equals(NIL_TF)) return null
-    else if (t.equals(FEED_T)) return decoder.feed(input)
-    else if (t.equals(MSG_T)) return decoder.message(input)
-    else if (t.equals(BLOB_T)) return decoder.blob(input)
-    else if (t.equals(BOX_T)) return decoder.box(input)
-    else if (t.equals(SIGNATURE_T)) {
-      if (tf.equals(SIGNATURE_TF)) return decoder.signature(input)
-      else throw new Error('Unknown signature type')
-    } else if (tf.equals(SIGNATURE_TF)) return decoder.signature(input)
-    else return input
-  } else if (typeof input === 'object' && input !== null) {
+  /* recurse */
+  if (Array.isArray(input)) return input.map(decode)
+  if (typeof input === 'object') {
+    // know it's not null, Array
     const output = {}
-    for (let key in input) {
-      const y = input[key]
-      const x = decode(y)
-      output[key] = x
+    for (const key in input) {
+      output[key] = decode(input[key])
     }
     return output
-  } // FIXME: more checks, including floats!
-  else return input
+  }
+
+  // FIXME: more checks, including floats!
+  throw new Error("don't know how to decode: " + input)
 }
 
 module.exports = {
   encode,
   decode,
-  bfeTypes: TYPES,
-  bfeNamedTypes: NAMED_TYPES,
+  toString: decode,
```

Comment:
can we leave decode here? I kind of like it :) Fine with adding an alias for toString

Reply:
yeah decode is still there :)